### ISSUE TITLE
Unhandled exception may hang the app when importing multiple books

### ISSUE THREAD

**Issue by naico-wang** [readest/readest#75]
## Bug Description

```
App hanged when import more than 50 epub files to library
```

## Expected behavior

```
All the files should be imported correctly
```

**Comment by chrox** [comment]
Any console log in the inspect tab of the app by right click on the page and go to the Inspect Element? And what is your operating system?

**Comment by naico-wang** [comment]
> Any console log in the inspect tab of the app by right click on the page and go to the Inspect Element? And what is your operating system?

<img width="621" alt="image" src="https://github.com/user-attachments/assets/fcb914f8-9834-4c91-aaaa-825ec62a89b2" />
My OS is MacOS 15.2 (24C101)

**Comment by naico-wang** [comment]
> Any console log in the inspect tab of the app by right click on the page and go to the Inspect Element? And what is your operating system?

Thanks for your reminding, I found that one of my epub file is broken and if I skip importing this file, everything works fine. I imported 200 files and they are good to go.

BTW, can we handle this kind of uncaught errors and popup a windows to allow user to know that one of the files is broken, and then skip the broken files, go on proceeding the other ones?

**Comment by chrox** [comment]
It seems one of your epub files is malformed. Can you add the files one by one and find the document that causes the problem? If it's possible can you send the document to my email? Thanks.

**Comment by chrox** [comment]
> Thanks for your reminding, I found that one of my epub file is broken and if I skip importing this file, everything works fine. I imported 200 files and they are good to go.
> 
> BTW, can we handle this kind of uncaught errors and popup a windows to allow user to know that one of the files is broken, and then skip the broken files, go on proceeding the other ones?

That will be reasonable.

**Comment by naico-wang** [comment]
> > Thanks for your reminding, I found that one of my epub file is broken and if I skip importing this file, everything works fine. I imported 200 files and they are good to go.
> > BTW, can we handle this kind of uncaught errors and popup a windows to allow user to know that one of the files is broken, and then skip the broken files, go on proceeding the other ones?
> 
> That will be reasonable.

I have emailed you the broken epub file, maybe can help you to find this issue.

Please check your gmail.

Thank you again for your quick response.

Really a good reader application :D

**Comment by chrox** [comment]
Fixed in #76
